janis_assistant/modifiers/base.py

```diff
@@ -34,3 +37,79 @@ def apply_many(
             wf, inputs = m.apply(wf, inputs or {}, hints)
 
         return wf, inputs
+
+
+class FileModifierBase(PipelineModifierBase):
+    def localise_inputs(
+        self,
+        inpid: str,
+        inptype: DataType,
+        dest_dir: str,
+        source: Union[str, List[str]],
+        localise_secondary_files: bool = True,
+    ):
+        if isinstance(source, list):
+            return [self.localise_inputs(inpid, inptype, dest_dir, s) for s in source]
+
+        fs = FileScheme.get_type_by_prefix(source)()
```

Comment:
If when using the FileFinder (eg: locating references when they're not specified), and the `source` is a local file (eg: somewhere on the local filesystem), it doesn't get localised to the current execution context (ie: same disk).

This isn't really a problem, Cromwell is going to probably put it in the `cached-copy` folder within the `janis/execution` directory anyway.